Source/Engine/Scripting/Types.h

```diff
@@ -49,6 +49,17 @@ typedef void MObject;
 #define USE_MONO_AOT_MODE MONO_AOT_MODE_NONE
 #endif
 
+#if USE_NETCORE
+struct _MonoDomain {};
+struct _MonoThread {};
+#endif
+
+#if USE_NETCORE
+typedef unsigned long long gchandle;
+#else
+typedef uint32 gchandle;
```

Comment:
Maybe we will simplfy gchandle to uint64 so Mono runtime will do casting but we will get simpler code in engine?